client/my-sites/pages/page-card-info/index.jsx

```diff
@@ -13,20 +13,62 @@ import Gridicon from 'gridicons';
  * Internal dependencies
  */
 import { isFrontPage, isPostsPage } from 'state/pages/selectors';
+import PostRelativeTimeStatus from 'my-sites/post-relative-time-status';
+import { canCurrentUser } from 'state/selectors';
+import { getEditorPath } from 'state/ui/editor/selectors';
 
-function PageCardInfo( { translate, moment, page, showTimestamp, isFront, isPosts, siteUrl } ) {
+const getContentLink = ( state, siteId, page ) => {
+	let contentLinkURL = page.URL;
+	let contentLinkTarget = '_blank';
+
+	if ( canCurrentUser( state, siteId, 'edit_pages' ) && page.status !== 'trash' ) {
+		contentLinkURL = getEditorPath( state, siteId, page.ID );
+		contentLinkTarget = null;
+	} else if ( page.status === 'trash' ) {
+		contentLinkURL = null;
+	}
+
+	return { contentLinkURL, contentLinkTarget };
+};
+
+function PageCardInfo( {
+	translate,
+	moment,
+	page,
+	showTimestamp,
+	isFront,
+	isPosts,
+	siteUrl,
+	contentLink,
+} ) {
 	const iconSize = 12;
+	const renderFutureTimestamp = function() {
+		if ( page.status === 'future' ) {
+			return (
+				<PostRelativeTimeStatus
+					post={ page }
+					link={ contentLink.contentLinkURL }
+					target={ contentLink.contentLinkTarget }
+				/>
+			);
+		}
+
+		return null;
+	};
 
 	return (
 		<div className="page-card-info">
 			{ siteUrl && <div className="page-card-info__site-url">{ siteUrl }</div> }
 			<div>
-				{ showTimestamp && (
-					<span className="page-card-info__item">
-						<Gridicon icon="time" size={ iconSize } className="page-card-info__item-icon" />
-						<span className="page-card-info__item-text">{ moment( page.modified ).fromNow() }</span>
-					</span>
-				) }
+				{ showTimestamp &&
+					( renderFutureTimestamp() || (
+						<span className="page-card-info__item">
+							<Gridicon icon="time" size={ iconSize } className="page-card-info__item-icon" />
+							<span className="page-card-info__item-text">
+								{ moment( page.modified ).fromNow() }
+							</span>
+						</span>
+					) ) }
```

Comment:
Prettier keeps reformatting this from:

``` js
{ showTimestamp && (
	renderFutureTimestamp() || (
		<span className="page-card-info__item">
			<Gridicon icon="time" size={ iconSize } className="page-card-info__item-icon" />
			<span className="page-card-info__item-text">
				{ moment( page.modified ).fromNow() }
			</span>
		</span>
	)
) }
```
to what is currently displayed. It looks a bit weird to me personally.